### ISSUE TITLE
Pass assigns? session?

### ISSUE THREAD

**Issue by ndrean** [dwyl/elixir-auth-github-demo#20]
If like me, you wish to get rid of this ugly URL after login, you need to change the `render` for a `redirect(to: "/welcome")` (where "welcome" can be any name you wish) and:
- add a new endpoint in the router, say `get "/welcome`
- create the corresponding new controller  "WelcomeController.ex", 
- add a view named after the controller, 
- add  a "sub-templates" folder also named like your controller
- finally, the template itself named "welcome.html.heex" is you use render("welcome.html") in the welcome controller.

When you [render](https://hexdocs.pm/phoenix/Phoenix.Controller.html#render/2) from a controller,  the `assigns` are available to the view. However, the URL remains unchanged.

When you [redirect](https://hexdocs.pm/phoenix/Phoenix.Controller.html#redirect/2), you **lose the conn**, thus the assigns !?
The only way (I found) to get what you want in the new controller is via the session.
Then if you want to use/render elements of the session in the template, you need to put the data back from the session into the conn.assigns and the template can now use them.

I believe this is correct after experimenting but I would be happy to get confirmation as all this is not obvious.
I am surprised that the conn is emptied when you redirect though.

**Comment by nelsonic** [comment]
Depends how you redirect. But yes, using `session` is your best bet.
It's why we have `auth_plug` https://github.com/dwyl/auth_plug

**Comment by ndrean** [comment]
I understand why the conn is lost after redirect: inspecting `self()` shows that the PID  changed, so you create a new process. Ok! Not evident! I will explore your `auth_plug`. I saw you used it in the [liveview chat](https://github.com/dwyl/phoenix-liveview-chat-example) but I skipped it in my first pass. Plenty, really plenty of how-to details to make this work. The material you provide is more than helpful, like gold :) Can only give one star ;)
